### PR TITLE
FIX: make sure all ticks show up for colorbar minor tick

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

#12095 shows that if you expect a minor colorbar tick right at the end of a colorbar it sometimes doesn't show up due to floating point precision issues.  This fixes that...

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- n/a New features are documented, with examples if plot related
-  n/a Documentation is sphinx and numpydoc compliant
-  n/a Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
-  n/a Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/colorbar.py [hunk 1]
```diff
@@ -267,7 +267,8 @@ def __call__(self):
         vmin = self._colorbar.norm.vmin
         vmax = self._colorbar.norm.vmax
         ticks = ticker.AutoMinorLocator.__call__(self)
-        return ticks[(ticks >= vmin) & (ticks <= vmax)]
+        rtol = (vmax - vmin) * 1e-10
+        return ticks[(ticks >= vmin - rtol) & (ticks <= vmax + rtol)]
```
Comment (timhoffm) [inline lib/matplotlib/colorbar.py]: Just as a thought: Can it happen, that vmin > vmax? If so, neither the original nor the fixed code would work in that case.

Comment (jklymak) [reply lib/matplotlib/colorbar.py]: Yes but that gets checked and swapped inside the method before this code

Comment (jklymak) [reply lib/matplotlib/colorbar.py]: Confirmed that vmax and vmin get flipped if they are specified out of order.   Which maybe is bad behaviour?  But its what colorbar does...

**Comment by jklymak** [comment]
marking WIP:

 1. needs tests
 2. I guess I think `MaxNLocator` should be doing this itself; i.e. it should not (by default) return ticks that are out of range.  Thats an API change though, and may break other things, so maybe beyond this PR...

**Comment by jklymak** [comment]
This is fine for now.  New test fails master, passes with this PR.  Yes, I could parameterize the test, but I don't think its so unwieldly as is. 

The more general issue of MaxNLocator can get its own PR where it can be properly debated.

**Comment by jklymak** [comment]
Doc build seems broken, but I don't think its this PR.  Something w/ stix fonts...

**Comment by Carreau** [comment]
> Something went wrong ... Please have a look at my logs.

When things like that happen, feel free to ping me. This was an actual bug that (should) be fixed.

**Comment by tacaswell** [comment]
@Carreau I would have gotten there eventually....Trying to get 3.0 finished off today :crossed_fingers: 

Could you also add a link to the logs in that error message?  The question I was going to ask you first was "remind me how to find the logs...."

**Comment by pharshalp** [comment]
I should have noticed this before but I forgot to add a reference to the newly introduced (in PR #11584 for 3.0) cbar.minorticks_on/off methods in the examples section (created a quick PR #12130 for that just now).

File: lib/matplotlib/colorbar.py [hunk 2]
```diff
@@ -267,7 +267,8 @@ def __call__(self):
         vmin = self._colorbar.norm.vmin
         vmax = self._colorbar.norm.vmax
         ticks = ticker.AutoMinorLocator.__call__(self)
```
Comment (peterkinalex) [inline lib/matplotlib/colorbar.py]: [Algorithm Implementation - Wikibooks, open books for an open world.pdf](https://github.com/matplotlib/matplotlib/files/2386084/Algorithm.Implementation.-.Wikibooks.open.books.for.an.open.world.pdf)

**Comment by Carreau** [comment]
> Could you also add a link to the logs in that error message? The question I was going to ask you first was "remind me how to find the logs...."

I can try, but not directly, as the link changes regularly because of authentication: 

https://dashboard.heroku.com/apps/meeseeksbot

Is the closest I can get for you, then click on papertrail.

The bug is (was?) in an optimisation I was doing of reusing previously cloned matplotlib, where tring to fetch a branch fails as it is already the checked-out branch. 

That was becoming necessary as matplotlib takes ~70 seconds to clone which is too close the timeout where the bot get killed.

**Pull request merged**